A4A: add signup multiple steps form

andrii-lysenko opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Resolves https://github.com/Automattic/automattic-for-agencies-dev/issues/1728

## Proposed Changes

This PR adds new Mutiple step contact form. The main purpose is to have a skeleton, so it's not 100% properly styled yet. Validation is disabled as well.

The main goal is to bootstrap all the steps and attach necessary logic. 

This also adds form for the first page.

<img width="654" alt="Screenshot 2025-01-30 at 5 06 04 PM" src="https://github.com/user-attachments/assets/9425e101-88e6-4c13-82db-b60422f45058" />


## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Using live link below navigate to `/signup/wc-asia`.
- Check multiple steps by clicking `Continue` button.


https://github.com/user-attachments/assets/90961ce9-a148-4c30-b395-49b8dcb4468a



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

andrii-lysenko committed:
A4A: add signup multiple steps form

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/contact-form/index.tsx:
```diff
@@ -0,0 +1,167 @@
+import FormPhoneInput from 'calypso/components/forms/form-phone-input';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import { useGetSupportedSMSCountries } from 'calypso/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-editor/hooks';
+import './style.scss';
```
Can we add an extra line before `import './style.scss';`. We have this throughout the app.

```suggestion

import './style.scss';
```

yashwin replied:
Can we add an extra line before `import './style.scss';`. We have this throughout the app.

```suggestion

import './style.scss';
```

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/contact-form/index.tsx:
```diff
@@ -0,0 +1,167 @@
+		>
+			<div className="signup-multi-step-form__fields">
+				<div className="signup-multi-step-form__name-fields">
+					<FormField label={ translate( 'Your first name' ) } isRequired>
```
Can we use `htmlFor` for all the fields on this form? 

We could use `FormLabel` that supports it or add the support on the `FormField` component

andrii-lysenko replied:
Thanks for noting that. Let's address this one with one of the page tasks, to avoid growing this PR. What do you think?

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/contact-form/index.tsx:
```diff
@@ -0,0 +1,167 @@
+
+				<FormField label={ translate( 'Phone number' ) } showOptionalLabel>
+					<FormPhoneInput
+						countrySelectProps={ {
+							'data-testid': 'a4a-signup-country-code-select',
+						} }
+						phoneInputProps={ {
+							'data-testid': 'a4a-signup-phone-number-input',
+						} }
```
We haven't added any test cases, and we already have the same `data-testid`s in the original signup form. Can we remove this for now or keep it different?

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/contact-form/style.scss:
```diff
@@ -0,0 +1,139 @@
+	padding: 2px;
+
+	&-label {
+		font-size: 0.875rem;
+		font-weight: 500;
```
Can we use A4A Typography for font styles in this file: https://github.com/Automattic/wp-calypso/blob/a5453358292e766c2064549e7987d5dfcd7a14db/client/assets/stylesheets/shared/mixins/_a4a-typography.scss? This will make it easy for us to migrate to Core Typography.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/contact-form/style.scss:
```diff
@@ -0,0 +1,139 @@
+	margin-bottom: 0;
+	padding: 2px;
+
+	&-label {
```
Can we avoid using `&-` in the file, and use `.a4a-form__section-field-label` instead? This will make it a lot easier to search.

jkguidaven replied:
I think we don't need to override this.. The existing form field component already matches the design here. https://github.com/Automattic/automattic-for-agencies-dev/issues/1729

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/contact-form/style.scss:
```diff
@@ -0,0 +1,139 @@
+	}
+
+	&::-webkit-scrollbar-track {
+		background: var(--studio-gray-0);
```
It is better to avoid variables from the studio everywhere on this file, and instead, we should use `--color-neutral`. This will ensure we are using the color based on the theme.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/index.tsx:
```diff
@@ -0,0 +1,44 @@
+			case 2:
+				return <SignupContactForm onContinue={ () => setCurrentStep( 3 ) } />;
+			case 3:
+				return <div>Finish</div>;
```
Should we translate this text?

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/style.scss:
```diff
@@ -0,0 +1,40 @@
+		margin-bottom: 40px;
+		flex-shrink: 0;
+
+		&-title {
```
Same as above, it's better to avoid `&-` and use A4A Typography.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/step-progress/style.scss:
```diff
@@ -0,0 +1,73 @@
+	width: 100%;
+	background: var(--studio-white);
+
+	&__steps {
```
Same as above, it's better to avoid `&__` and use A4A Typography

yashwin reviewed (commented):
Thanks for working on this!

I noticed the background on the video added here is white, whereas it shows a dark background here:

<img width="1693" alt="Screenshot 2025-02-03 at 12 56 08 PM" src="https://github.com/user-attachments/assets/e2ea1a65-57e7-4ade-b17b-cdf9acccf11d" />

I also left some comments, let me what you think!

jkguidaven commented on client/a8c-for-agencies/sections/signup/signup-v2/components/step-progress/style.scss:
```diff
@@ -0,0 +1,73 @@
+.step-progress {
+	width: 100%;
+	background: var(--studio-white);
```
We should probably avoid directly referencing `--studio-*` but instead use the equivalent theme variable. `--color-surface`

jkguidaven commented on client/a8c-for-agencies/sections/signup/signup-v2/components/step-progress/style.scss:
```diff
@@ -0,0 +1,73 @@
+
+		&:not(.is-active):not(.is-complete) {
+			.step-progress__step-indicator {
+				background-color: var(--studio-gray-5);
```
As I mentioned above. We can replace `--studio-gray-*` here with `--color-neutral-*`

andrii-lysenko commented:
@yashwin @jkguidaven Thank you for the reviews. I addressed some of the comments like usage of `preventWindows` and removing `&__` from styles replacing with the full name. As per others, I suggest addressing these on `per page` basis, as we will improve every page styles and code. What do you think? These include also using `htmlFor` as it probably require changing the component outside the form itself, so I would like not to grow this PR even more.

I've also change base of this PR to `trunk` and will close https://github.com/Automattic/wp-calypso/pull/99137 to avoid problems with merging.

andrii-lysenko commented:
Closing this PR to avoid confusion. All the changes will be delivered from this PR: https://github.com/Automattic/wp-calypso/pull/99186

I will take notes on unresolved suggestions on the appropriate tasks. Thank you!
